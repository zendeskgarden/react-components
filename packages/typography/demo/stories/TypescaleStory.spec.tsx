import React from 'react';
import { render } from 'garden-test-utils';
import 'jest-styled-components';
import { TypescaleStory } from './TypescaleStory';

const renderAndMatchSnapshot = (props: any) => {
  const { container } = render(<TypescaleStory {...props} />);
  expect(container.firstChild).toMatchSnapshot();
};

describe('TypescaleStory Component', () => {
  it('renders default TypescaleStory (medium size)', () => {
    renderAndMatchSnapshot({});
  });

  it('renders TypescaleStory with small size', () => {
    renderAndMatchSnapshot({ size: 'small' });
  });

  it('renders TypescaleStory with large size', () => {
    renderAndMatchSnapshot({ size: 'large' });
  });

  it('renders TypescaleStory with extra-large size', () => {
    renderAndMatchSnapshot({ size: 'extra-large' });
  });

  it('renders TypescaleStory with 2x-large size', () => {
    renderAndMatchSnapshot({ size: '2x-large' });
  });

  it('renders TypescaleStory with 3x-large size', () => {
    renderAndMatchSnapshot({ size: '3x-large' });
  });

  it('renders TypescaleStory with display name', () => {
    renderAndMatchSnapshot({ hasDisplayName: true });
  });

  it('renders TypescaleStory with bold text', () => {
    renderAndMatchSnapshot({ isBold: true });
  });

  it('renders TypescaleStory with monospace font', () => {
    renderAndMatchSnapshot({ isMonospace: true });
  });

  it('renders TypescaleStory with a custom tag', () => {
    renderAndMatchSnapshot({ tag: 'span' });
  });

  it('renders TypescaleStory with small size and display name', () => {
    renderAndMatchSnapshot({ size: 'small', hasDisplayName: true });
  });

  it('renders TypescaleStory with large size and bold text', () => {
    renderAndMatchSnapshot({ size: 'large', isBold: true });
  });

  it('renders TypescaleStory with extra-large size and monospace font', () => {
    renderAndMatchSnapshot({ size: 'extra-large', isMonospace: true });
  });

  it('renders TypescaleStory with 2x-large size and custom tag', () => {
    renderAndMatchSnapshot({ size: '2x-large', tag: 'h1' });
  });

  it('renders TypescaleStory with 3x-large size, display name, and bold text', () => {
    renderAndMatchSnapshot({ size: '3x-large', hasDisplayName: true, isBold: true });
  });

  it('renders TypescaleStory with small size, monospace font, and custom tag', () => {
    renderAndMatchSnapshot({ size: 'small', isMonospace: true, tag: 'code' });
  });

  it('renders TypescaleStory with large size, display name, bold text, and monospace font', () => {
    renderAndMatchSnapshot({
      size: 'large',
      hasDisplayName: true,
      isBold: true,
      isMonospace: true
    });
  });

  it('renders TypescaleStory with extra-large size, display name, bold text, and custom tag', () => {
    renderAndMatchSnapshot({
      size: 'extra-large',
      hasDisplayName: true,
      isBold: true,
      tag: 'strong'
    });
  });

  it('renders TypescaleStory with 2x-large size, display name, monospace font, and custom tag', () => {
    renderAndMatchSnapshot({
      size: '2x-large',
      hasDisplayName: true,
      isMonospace: true,
      tag: 'pre'
    });
  });

  it('renders TypescaleStory with 3x-large size, display name, bold text, monospace font, and custom tag', () => {
    renderAndMatchSnapshot({
      size: '3x-large',
      hasDisplayName: true,
      isBold: true,
      isMonospace: true,
      tag: 'h2'
    });
  });
});
