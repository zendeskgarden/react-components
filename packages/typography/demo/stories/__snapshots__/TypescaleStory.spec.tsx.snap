// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypescaleStory Component renders TypescaleStory with 2x-large size 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 32px;
  font-size: 26px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with 2x-large size and custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 32px;
  font-size: 26px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<h1
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with 2x-large size, display name, monospace font, and custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 32px;
  font-size: 26px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<pre
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;XXL&gt;
  &lt;/XXL&gt;
</pre>
`;

exports[`TypescaleStory Component renders TypescaleStory with 3x-large size 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 44px;
  font-size: 36px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with 3x-large size, display name, and bold text 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 44px;
  font-size: 36px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;XXXL&gt;
  &lt;/XXXL&gt;
</div>
`;

exports[`TypescaleStory Component renders TypescaleStory with 3x-large size, display name, bold text, monospace font, and custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 44px;
  font-size: 36px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<h2
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;XXXL&gt;
  &lt;/XXXL&gt;
</h2>
`;

exports[`TypescaleStory Component renders TypescaleStory with a custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 20px;
  font-size: 14px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<span
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with bold text 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 20px;
  font-size: 14px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with display name 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 20px;
  font-size: 14px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;MD&gt;
  &lt;/MD&gt;
</div>
`;

exports[`TypescaleStory Component renders TypescaleStory with extra-large size 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 28px;
  font-size: 22px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with extra-large size and monospace font 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 28px;
  font-size: 22px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with extra-large size, display name, bold text, and custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 28px;
  font-size: 22px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<strong
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;XL&gt;
  &lt;/XL&gt;
</strong>
`;

exports[`TypescaleStory Component renders TypescaleStory with large size 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 24px;
  font-size: 18px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with large size and bold text 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 24px;
  font-size: 18px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with large size, display name, bold text, and monospace font 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 23px;
  font-family: SFMono-Regular,Consolas,"Liberation Mono",Menlo,Courier,monospace;
  font-size: 17px;
  font-weight: 600;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;LG&gt;
  &lt;/LG&gt;
</div>
`;

exports[`TypescaleStory Component renders TypescaleStory with monospace font 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 19px;
  font-family: SFMono-Regular,Consolas,"Liberation Mono",Menlo,Courier,monospace;
  font-size: 13px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with small size 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 16px;
  font-size: 12px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders TypescaleStory with small size and display name 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 16px;
  font-size: 12px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
>
  &lt;SM&gt;
  &lt;/SM&gt;
</div>
`;

exports[`TypescaleStory Component renders TypescaleStory with small size, monospace font, and custom tag 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 15px;
  font-family: SFMono-Regular,Consolas,"Liberation Mono",Menlo,Courier,monospace;
  font-size: 11px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<code
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;

exports[`TypescaleStory Component renders default TypescaleStory (medium size) 1`] = `
.c0 {
  transition: color 0.1s ease-in-out;
  line-height: 20px;
  font-size: 14px;
  font-weight: 400;
  direction: ltr;
}

.c0[hidden] {
  display: inline;
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div
  class="c0"
  data-garden-id="typography.font"
  data-garden-version="9.2.0"
/>
`;
